ExampleApp Android/src/main/java/com/rootstrap/flowforms/example/SignUpFormFragment.kt

```diff
@@ -0,0 +1,139 @@
+        binding?.let {
+            it.lifecycleOwner = viewLifecycleOwner
+            it.formModel = viewModel.formModel
+            it.continueButton.setOnClickListener {
+                Toast.makeText(requireContext(), R.string.account_registered, Toast.LENGTH_SHORT).show()
+            }
+        }
+        listenStatusChanges()
+        bindFields()
+    }
+
+    private fun listenStatusChanges() {
+        viewModel.form.fields.value.let {
+            repeatOnLifeCycleScope(
+                { it[SignUpFormModel.NAME]?.status?.collect(::onNameStatusChange) },
+                { it[SignUpFormModel.EMAIL]?.status?.collect(::onEmailStatusChange) },
+                { it[SignUpFormModel.NEW_PASSWORD]?.status?.collect(::onPasswordStatusChange) },
+                { it[SignUpFormModel.CONFIRM_PASSWORD]?.status?.collect(::onConfirmPasswordChange) },
+                { viewModel.form.status.collect(::onFormStatusChange) }
+            )
+        }
+    }
+
+    private fun bindFields() {
+        binding?.apply {
+            viewModel.form.bind(lifecycleScope,
+                nameInputEditText to SignUpFormModel.NAME,
+                emailInputEditText to SignUpFormModel.EMAIL,
+                passwordInputEditText to SignUpFormModel.NEW_PASSWORD,
+                confirmPasswordInputEditText to SignUpFormModel.CONFIRM_PASSWORD
+            )
+            viewModel.form.bind(this@SignUpFormFragment, lifecycleScope,
+                viewModel.formModel.confirm to SignUpFormModel.CONFIRMATION
+            )
+        }
+    }
+
+    private fun onNameStatusChange(status: FieldStatus) {
+        binding?.apply {
+            when (status.code) {
+                CORRECT, UNMODIFIED -> nameInputLayout.error = null
+                else -> nameInputLayout.error = getString(R.string.required_field)
+            }
+        }
+    }
+
+    private fun onEmailStatusChange(status: FieldStatus) {
+        binding?.apply {
+            when (status.code) {
+                CORRECT, UNMODIFIED -> emailInputLayout.error = null
+                INVALID_EMAIL -> emailInputLayout.error = getString(R.string.invalid_email)
+                else -> emailInputLayout.error = getString(R.string.required_field)
+            }
+        }
+    }
+
+    private fun onPasswordStatusChange(status: FieldStatus) {
+        binding?.apply {
+            when (status.code) {
+                CORRECT, UNMODIFIED -> passwordInputLayout.error = null
+                MIN_LENGTH_UNSATISFIED -> passwordInputLayout.error = getString(R.string.min_length,
+                    MIN_PASSWORD_LENGTH
+                )
+                else -> passwordInputLayout.error = getString(R.string.required_field)
+            }
+        }
+    }
+
+    private fun onConfirmPasswordChange(status: FieldStatus) {
+        binding?.apply {
+            when (status.code) {
+                CORRECT, UNMODIFIED -> confirmPasswordInputLayout.error = null
```

Comment:
it seems like  CORRECT and UNMODIFIED are the same in all cases

Reply:
Yes, I updated this implementation logic in #30, so we directly handle these in the `else` case